sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_policy_token_header_async.py

```diff
@@ -0,0 +1,103 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from typing import TypeVar, Any, TYPE_CHECKING
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+
+from azure.core.pipeline.policies import AsyncHTTPPolicy
+from azure.core.rest import HttpRequest, AsyncHttpResponse
+from azure.core.pipeline.transport import (
+    AsyncHttpResponse as LegacyAsyncHttpResponse,
+    HttpRequest as LegacyHttpRequest,
+)
+from ._policy_token_header import _create_acquire_policy_request, _update_request_with_policy_token
+from ._authentication_async import await_result
+
+if TYPE_CHECKING:
+    from .._async_pipeline_client import AsyncARMPipelineClient
+
+AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType", AsyncHttpResponse, LegacyAsyncHttpResponse)
+HTTPRequestType = TypeVar("HTTPRequestType", HttpRequest, LegacyHttpRequest)
+
+
+class AsyncPolicyTokenHeaderPolicy(AsyncHTTPPolicy[HTTPRequestType, AsyncHTTPResponseType]):
+    """Async HTTP pipeline policy for adding policy token headers to Azure Resource Manager requests.
+
+    This policy handles the acquisition and application of Azure Policy tokens for external
+    policy evaluation. It integrates with the Azure Policy service to obtain tokens that
+    enable policy evaluation during resource operations.
+
+    :param client: The async ARM pipeline client used for making policy token requests
+    :type client: ~azure.mgmt.core.AsyncARMPipelineClient
+    :param kwargs: Additional keyword arguments passed to the base policy
+    :type kwargs: Any
+    """
+
+    def __init__(self, client: "AsyncARMPipelineClient", **kwargs: Any) -> None:  # pylint: disable=unused-argument
+        """Initialize the async policy token header policy.
+
+        :param client: The async ARM pipeline client used for making policy token requests
+        :type client: ~azure.mgmt.core.AsyncARMPipelineClient
+        :param kwargs: Additional keyword arguments passed to the base policy
+        :type kwargs: Any
+        """
+        self._client = client
+
+    async def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
+        """Process the request to add policy token headers if needed.
+
+        This method is called for each outgoing request. It checks if a policy token
+        is needed and if so, acquires one and adds it to the request headers.
+
+        :param request: The pipeline request to process
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        acquire_policy_token = request.context.options.pop("acquire_policy_token", False)
```

Comment:
Are we expecting customers to pass in this flag when they call a method?
If this is a customer facing feature, I think we should name it something more related to the usage scenario, like `"evaluate_policy"`.

Otherwise, if this is a fully internal flag then it doesn't really matter.

Reply:
(1) Yeah, most of SDK users won't use this policy. So I design to make the policy work only when SDK users pass the special flag in calling a method. 
(2) About flag name, I try to make the flag name keep consistent with CLI https://github.com/Azure/azure-cli/pull/31741/files#diff-3f5f2f988d8e28f084ce7df08a3ffc5f1faeb101490f837d14cc6c5a5d9a34e4 and original REST API definition https://github.com/Azure/azure-rest-api-specs/blob/c2f167ec164a2349235d8aee1b8189b883ec93c6/specification/resources/resource-manager/Microsoft.Authorization/stable/2025-03-01/policyTokens.json#L37